google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/Query.java

```diff
@@ -297,6 +321,73 @@ private static ByteString wrapKey(String key) {
     return ByteString.copyFromUtf8(key);
   }
 
+  /**
+   * A Query Paginator that will split a query into small chunks. See {@link
+   * Query#createPaginator(int)} for example usage.
+   */
+  @BetaApi("This surface is stable yet it might be removed in the future.")
+  public static class QueryPaginator {
+
+    private final boolean hasOverallLimit;
+    private long remainingRows;
+    private Query query;
+    private final int pageSize;
+    private ByteString prevSplitPoint;
+
+    QueryPaginator(@Nonnull Query query, int pageSize) {
+      this.hasOverallLimit = query.builder.getRowsLimit() == 0 ? false : true;
```

Comment:
this.hasOverallLimit = query.builder.getRowsLimit() > 0;

Reply:
🤦‍♀️